components/LocationInput.tsx

```diff
@@ -225,8 +224,8 @@ const LocationInput: FC<LocationInputProps> = ({
   ]);
 
   useEffect(() => {
-    setIsSubmissionEnabled(isInUK && isPastZoomThreshold);
-  }, [isInUK, isPastZoomThreshold, setIsSubmissionEnabled]);
+    setIsSubmissionEnabled(isInUK && isPastZoomThreshold && !isMoving);
+  }, [isInUK, isPastZoomThreshold, setIsSubmissionEnabled, isMoving]);
```

Comment:
Is there a reason we are prop drilling setState functions? I feel like a callback design pattern makes more sense and is much more extensible. Let the parent handle the state updates in it's own file and simply pass in a call back that passes back the information that your parent needs to update the state.